xml/vk.xml

```diff
@@ -154,16 +154,37 @@ server.
 
         <type category="define" name="VK_DEFINE_NON_DISPATCHABLE_HANDLE">
 #if !defined(VK_DEFINE_NON_DISPATCHABLE_HANDLE)
-#if defined(__LP64__) || defined(_WIN64) || (defined(__x86_64__) &amp;&amp; !defined(__ILP32__) ) || defined(_M_X64) || defined(__ia64) || defined (_M_IA64) || defined(__aarch64__) || defined(__powerpc64__)
+#if defined(__cplusplus) &amp;&amp; ( __cplusplus >= 201100 || _MSC_VER >= 1800)
+        #define VK_DEFINE_NON_DISPATCHABLE_HANDLE(object) \
+            class object{ \
+            private: \
+                uint64_t h; \
+            public: \
+                object(){}; \
```

Comment:
The Vulkan WG had some code very similar to this in the header before 1.0 was released, but we ultimately ditched it because it introduced too many problems.

One specific bug I recall is that this default constructor needs to initialize h to zero. This is the constructor that gets called when the handle is initialized by aggregate initialization, so leaving it uninitialized here breaks code that uses the following style to zero-init structures:

// renderPass and framebuffer would no longer be zero-initialized after this change
VkRenderPassBeginInfo beginInfo = { VK_STRUCTURE_TYPE_RENDER_PASS_BEGIN_INFO };

This specific problem is obviously fixable, but nonetheless I don't think the WG will want to make a change like this.

Reply:
Not so obviously fixable. That is if I don't want unnecessarily initialized handles.
That would mean I have to restrict it to aggregate type (forgo non-defaulted constructors), and restrict it to C++14 (that should force aggregate initialization, instead of value initialization... I think).